### PR TITLE
Create Perl tests of inspector behavior when mg returns a value

### PR DESCRIPTION
**Pull request by FinnitoProductions**
The response inspector flag parsing logic behaves incorrectly when a meta-get response includes a value. 

Introduce tests of this behavior for both standard requests and `mcp.internal` requests. These tests currently fail, but they are expected to pass once the bug is patched.

### CODE REVIEW

**Comment by dormando** [comment]
Noting for myself:
- mcmc tokenizer is being passed an incorrect mstart parameter for VA responses (is 1, should be 2), so it ends up looking at the second argument as though it's a flag and fails validation.
- from the code this looks like a pretty deep mistake, since we have to look at the result a bit before knowing where mstart is. it's unit tested but I did the mstart correctly since that was the lowest layer... sigh.

not immediately obvious how to fix simply so I'll come back to this after a break. forget where I put the HD/VA/etc code

**Comment by dormando** [comment]
More notes for me:

- can we call `mcmc_tokenize_res` against `r->resp.rline` instead since that will always point to the right place?
- if not, will have to have the parser remember the metastart offset instead of passing it into `mcmc_tokenize_res`

object vlen of 0 still has an internal vlen of 2, so I can just check for vlen != 0 to see where mstart is.

**Comment by dormando** [comment]
It's not a complicated fix but I need to make some API decisions. will get back to you.

**Comment by dormando** [comment]
Simplest method here might be:
- change mstart arg from offset to boolean
- if flag parsing fails, check if character is numeric and skip if so

that should be it. no need to check for HD vs VA anywhere up the chain.

**Comment by dormando** [comment]
The biggest issue here is:

- We tend to parse things directly out of input buffers.
- We then sometimes memcpy that actual request/response data into other buffers
- Now there are pointers in the `mcmc_resp_t` struct that point into the original buffer
- Thus we need to re-parse things sometimes. Which I'd prefer to not do.
- This matters because I can't just parse the meta line by looking at the `rline` offset without first re-parsing the thing.

I need to change the various pointers in `mcmc_resp_t` into offsets... been wanting to do this for a long time but it's a little more involved so I keep pushing it off.

TL;DR: I'm going to do the workaround of _not_ throwing an error if we run into numbers, and see about raising the priority of doing a large rework of mcmc.

**Comment by dormando** [comment]
fix should be merged now. Your tests + some additional ones I added all pass. Thanks!
